flows/references.go

```diff
@@ -93,18 +93,26 @@ func NewVariableLabelReference(nameMatch string) *LabelReference {
 // Validation
 //------------------------------------------------------------------------------------------
 
+// GroupReferenceValidation validates that the given group reference is either a concrete
+// reference or a name matcher
 func GroupReferenceValidation(sl validator.StructLevel) {
 	ref := sl.Current().Interface().(GroupReference)
-	if len(ref.UUID) != 0 && len(ref.NameMatch) != 0 {
-		sl.ReportError(ref.UUID, "UUID", "uuid", "uuidornamematch", "")
-		sl.ReportError(ref.NameMatch, "NameMatch", "name_match", "uuidornamematch", "")
+	if !xor(string(ref.UUID), ref.NameMatch) {
```

Comment:
`if (ref.UUID && ref.NameMatch) || (!ref.UUID && !ref.NameMatch)` ?

Not sure that's that much better, maybe we just need to be smarter? Or have a utility function which is `NeitherOrBoth`?

Reply:
strings don't have implicit boolean values so you have to wrap everything in len(..) > 0... so I think a utility function is still helpful here